### ISSUE TITLE
OpenTSDB Telnet put without tags requires space after value?

### ISSUE THREAD

**Issue by thomas725** [VictoriaMetrics/VictoriaMetrics#3290]
**Describe the bug**
Hi there!

I've taken the sample call
```
echo "put foo.bar.baz `date +%s` 123 tag1=value1 tag2=value2" | nc -N localhost 4242
```
From your docs @ https://docs.victoriametrics.com/Single-server-VictoriaMetrics.html#sending-data-via-telnet-put-protocol and removed the tags, which left me with:
```
echo "put foo.bar.baz `date +%s` 123" | nc -N localhost 4242
```
This doesn't work. I've found for it to work as intended, a space is required after the value. Like that:
```
echo "put foo.bar.baz `date +%s` 123 " | nc -N localhost 4242
```

Is this intended behavior, or a bug that could be fixed?

**Logs**
Ah, good point. Yes, it actually logs an error when calling the non functional line without space after the value:
```
error        VictoriaMetrics/lib/protoparser/opentsdb/parser.go:136        cannot unmarshal OpenTSDB line "put foo.bar.baz 1667207326 123": cannot find whitespace between value and the first tag in "foo.bar.baz 1667207326 123"
```

**Version**
```
$ ./victoria-metrics-prod --version
victoria-metrics-20221029-004923-tags-v1.83.0-0-g740f7ac5e
```

**Used command-line flags**
```
 -storageDataPath /opt/VictoriaMetrics/data -retentionPeriod=999y -search.maxPointsPerTimeseries=999999 -opentsdbListenAddr=:4242
```

**Comment by hagen1778** [comment]
Technically, it is not a bug. According to http://opentsdb.net/docs/build/html/api_telnet/put.html:

> "At least one tag pair must be present. Additional tag pairs can be added with spaces in between."

Do you use some standard openTSDB client or working on something custom?

**Comment by thomas725** [comment]
I've just written a custom bash script to get some testdata into VictoriaMetrics:
``` bash
while true
do
  ts=`date +%s`
  x=`cat /sys/class/thermal/thermal_zone0/temp` # ; cpu=$((x/1000)).${x: -3}
  echo "put pi4.temperature.cpu $ts $((x/1000)).${x: -3} "
  x=`vcgencmd measure_temp` # ; gpu=${x:5: -2}
  echo "put pi4.temperature.gpu $ts ${x:5: -2} "
  x=`vcgencmd measure_temp pmic` # ; pmi=${x:5: -2}
  echo "put pi4.temperature.pmi $ts ${x:5: -2} "
  x=`vcgencmd read_ring_osc` ; x=${x##* (} ; osc=${x:: -3}
  echo "put pi4.temperature.osc $ts ${x:: -3} "
  sleep 1
done | nc -vuN localhost 4242
#done |tee >(cat 1>&2) | nc -vuN localhost 4242 #switch to this for debugging
```

I'm currently evaluating alternatives to migrate my smart home data storage to, since I'm facing performance problems with my IO-Broker+MariaDB setup.

Over the last few weeks I've considered switching my IO-Broker datastorage from MariaDB to InfluxDB, but since I now found out about VictoriaMetrics this is my new favorite, since it feels much snappier and cleaner.

So the required extra space is not really a problem now that I've found out about it, it was just an anoying strange thing that took me about an hour to find. I'm not really experienced with any of those timeseries database related projects and protocols as of yet, still in the learning phase.

**Comment by thomas725** [comment]
I've now found that there's another even simpler interface to submit data through, where I don't even have to send the current timestamp along with the data: The Graphite plaintext protocol.

So I've adapted my test-data script to look like this now:
``` bash
while true
do
  x=`cat /sys/class/thermal/thermal_zone0/temp` # ; cpu=$((x/1000)).${x: -3}
  echo "pi4.temperature.cpu; $((x/1000)).${x: -3}"
  x=`vcgencmd measure_temp` # ; gpu=${x:5: -2}
  echo "pi4.temperature.gpu; ${x:5: -2}"
  x=`vcgencmd measure_temp pmic` # ; pmi=${x:5: -2}
  echo "pi4.temperature.pmi; ${x:5: -2}"
  x=`vcgencmd read_ring_osc` ; x=${x##* (} # ; osc=${x:: -3}
  echo "pi4.temperature.osc; ${x:: -3}"
  sleep 1
done | nc -vuN localhost 2003
```

**Comment by hagen1778** [comment]
> it would be nice if VictoriaMetrics would reply to unsuccessful put commands with error messages to the client sending the data, instead of only logging into it's own stdout log.

VictoriaMetrics parses request in a stream fashion. One request can contain N lines, and if some of them are incorrect - the rest will be accepted and stored. Because of this, errors are processed in a different fashion, so you don't get the error in response.

**Comment by valyala** [comment]
VictoriaMetrics should accept OpenTSDB put line without tags and without trailing whitespace starting from [v1.83.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.

**Comment by valyala** [comment]
FYI, this bugfix has been included in [VictoriaMetrics v1.79.5 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.5).
